pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -60,6 +60,12 @@ func (r *pullthroughBlobStore) remoteStat(ctx context.Context, dgst digest.Diges
 		return distribution.Descriptor{}, err
 	}
 
+	r.pullFromInsecureRegistries = false
```

Comment:
> If it's no longer useful for pullthrough tests then please remove it and use local variable instead.

Yes, change this variable is useless. I fixed the tests. ￼I did not remove a single test. It works without using this global variable. Why do you consider this a problem?